ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/PaywallDialog.kt

```diff
@@ -0,0 +1,80 @@
+package com.revenuecat.purchases.ui.revenuecatui
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Scaffold
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.ReadOnlyComposable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.window.Dialog
+import androidx.compose.ui.window.DialogProperties
+import androidx.window.core.layout.WindowWidthSizeClass
+import com.revenuecat.purchases.Purchases
+import com.revenuecat.purchases.PurchasesException
+import com.revenuecat.purchases.awaitCustomerInfo
+import com.revenuecat.purchases.ui.revenuecatui.helpers.Logger
+import com.revenuecat.purchases.ui.revenuecatui.helpers.computeWindowWidthSizeClass
+import kotlinx.coroutines.launch
+
+@Composable
+fun PaywallDialog(
+    paywallDialogOptions: PaywallDialogOptions,
+) {
+    val shouldDisplayBlock = paywallDialogOptions.shouldDisplayBlock
+    var shouldDisplayDialog by remember { mutableStateOf(shouldDisplayBlock == null) }
+    if (shouldDisplayBlock != null) {
+        LaunchedEffect(paywallDialogOptions) {
+            launch {
+                shouldDisplayDialog = try {
+                    // TODO-PAYWALLS: This won't receive updates in case the customer info changes and starts/stops
+                    // being eligible to display the paywall dialog. We would need to support multiple customer info
+                    // listeners to refresh this.
```

Comment:
Yeah that's how it works in iOS. We just want to do one initial check. I'd remove the TODO, since we wouldn't want to get notified of customer info changes and hide the paywall.